experimental/rockbuilder/rockbuilder.py

```diff
@@ -445,18 +468,36 @@ def do_therock(prj_builder):
 time.sleep(1)
 
 if args.project == "all":
+    if args.src_dir:
+        print(
+            '\nError, "--src-dir" parameter requires also to specify the project with the "--project"-parameter'
+        )
+        print('Alternatively you could use the "--src-base-dir" parameter.')
+        print("")
+        sys.exit(1)
```

Comment:
Could also use `parser.error()` here: https://docs.python.org/3/library/argparse.html#exiting-methods

See also the discussion on https://stackoverflow.com/questions/19414060/argparse-required-argument-y-if-x-is-present for other code patterns